benchmarks/misc/tests/assorted/misc-apply-array-headroom.js

```diff
@@ -0,0 +1,50 @@
+// Since bug 1108290 the JIT optimizes fun.apply(..., array) by taking
+// a fast inline path when the array is without holes and its length
+// equals its initialized length.
+//
+// This program applies arrays whose length is larger than their
+// initialized length, thus forcing the bailout path.
+//
+// Contrast the running time of this program with the running time of
+// misc-apply-array.js, which applies arrays whose length equals their
+// initialized length.  That program should run in 1/10 to 1/40 the
+// time of this one.
+
+function assertEq(result, expected) {
+  if (result !== expected)
+    throw "Assertion: Expected " + expected + ", got " + result;
+}
+
+var iter = 10000;
```

Comment:
Can I lower this to 1000? This decreases the time running from 2s to 300ms. Since this is running on multiple platforms and engines I prefer to have timings under 1s.